backend/dataall/core/environment/cdk/environment_stack.py

```diff
@@ -165,30 +165,44 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
                 f'arn:aws:iam::{self._environment.AwsAccountId}:role/{self.pivot_role_name}',
             )
 
-        # Environment S3 Bucket
-        default_environment_bucket = s3.Bucket(
+        # Environment Logging S3 Bucket
+        default_environment_log_bucket = s3.Bucket(
             self,
-            'EnvironmentDefaultBucket',
-            bucket_name=self._environment.EnvironmentDefaultBucketName,
+            'EnvironmentDefaultLogBucket',
+            bucket_name=self._environment.EnvironmentLogsBucketName,
             encryption=s3.BucketEncryption.S3_MANAGED,
             removal_policy=RemovalPolicy.RETAIN,
             block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
             versioned=True,
             enforce_ssl=True,
         )
-        default_environment_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
-        self.default_environment_bucket = default_environment_bucket
-
-        default_environment_bucket.add_to_resource_policy(
+        default_environment_log_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
+        default_environment_log_bucket.add_to_resource_policy(
             iam.PolicyStatement(
                 sid='AWSLogDeliveryWrite',
                 effect=iam.Effect.ALLOW,
                 principals=[iam.ServicePrincipal('logging.s3.amazonaws.com')],
                 actions=['s3:PutObject', 's3:PutObjectAcl'],
-                resources=[f'{default_environment_bucket.bucket_arn}/*'],
+                resources=[f'{default_environment_log_bucket.bucket_arn}/*'],
             )
         )
 
+        # Environment S3 Bucket
+        default_environment_bucket = s3.Bucket(
+            self,
+            'EnvironmentDefaultBucket',
+            bucket_name=self._environment.EnvironmentDefaultBucketName,
+            encryption=s3.BucketEncryption.S3_MANAGED,
+            removal_policy=RemovalPolicy.RETAIN,
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
+            versioned=True,
+            enforce_ssl=True,
+            server_access_logs_bucket=default_environment_log_bucket,
+            server_access_logs_prefix=f'access_logs/{self._environment.EnvironmentDefaultBucketName}/',
+        )
+        default_environment_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
```

Comment:
same as above, isn't the removal policy already defined in line 196

Reply:
same response - need to explicitly retain associated resources of bucket (i.e. bucket policy)